### ISSUE TITLE
Rollup v3 support

### ISSUE THREAD

**Issue by Mister-Hope** [Swatinem/rollup-plugin-dts#226]
https://github.com/rollup/rollup/tree/v3.0.0

```
├─┬ rollup-plugin-dts 4.2.3
│ └── ✕ unmet peer rollup@^2.55: found 3.0.0
```

```
packages/shared build: ./src/node/index.ts → ./lib/node/index.d.ts...
packages/shared build: (!) The "output.namespaceToStringTag" option is deprecated. Use the "output.generatedCode.symbols" option instead.
packages/shared build: [!] RollupError: Invalid value false for option "output.interop" - use one of "compat", "auto", "esModule", "default", "defaultOnly".
packages/shared build: https://rollupjs.org/guide/en/#outputinterop
packages/shared build: RollupError: Invalid value false for option "output.interop" - use one of "compat", "auto", "esModule", "default", "defaultOnly".
```

**Comment by Swatinem** [comment]
ah yes, the peer dependency says that the plugin is not (yet) compatible with rollup 3. Which is brand new I see ;-)

I believe node/npm changed the way they treat peerDependencies a while back, I will have to dig into how to properly use that.

**Comment by SukkaW** [comment]
Rollup v3 also deprecates several options:

- `namespaceToStringTag` is replaced by the `output.generatedCode.symbols`:

https://github.com/Swatinem/rollup-plugin-dts/blob/85524fa3706915e6f975e536bdfe8528a236b44b/src/transform/index.ts#L71

- `output.interop` no longer supports boolean type:

https://github.com/Swatinem/rollup-plugin-dts/blob/85524fa3706915e6f975e536bdfe8528a236b44b/src/transform/index.ts#L70

**Comment by pi0** [comment]
There seems also some configuration updates.


A really really ugly workaround for rollup@3 support:

```ts
    const dtsPlugin = dts(ctx.options.rollup.dts)
    rollupOptions.plugins.push({
      ...dtsPlugin,
      outputOptions (...args) {
        const opts = dtsPlugin.outputOptions(...args)
        opts.interop = 'esModule'
        delete opts.namespaceToStringTag
        opts.generatedCode = { symbols: false, ...opts.generatedCode }
        return opts
      }
    })
```

**Comment by nicolas377** [comment]
There isn't really a workaround needed for v3 support for right now. Rollup only reports a warning, and the outputted code isn't affected. Obviously, the PR should still be merged. Just putting this out there for onlookers.

**Comment by Mister-Hope** [comment]
> There isn't really a workaround needed for v3 support for right now. Rollup only reports a warning, and the outputted code isn't affected. Obviously, the PR should still be merged. Just putting this out there for onlookers.

why am I unable to have it work? From the log I provided, rollup gives an error. (I haven’t tested with later v3 yet, is this changed?)

**Comment by nicolas377** [comment]
That's weird. I used rollup-plugin-dts 4.2.3 with rollup 3.1.0 and `npm i --force` and it worked just fine. i'll check up on my package-lock and edit this in a moment

**Comment by Swatinem** [comment]
Just published `5.0.0` which supports/requires rollup 3, and also raises the minimum supported node version to 14.
